src/query/service/src/interpreters/interpreter_cluster_key_alter.rs

```diff
@@ -62,6 +62,8 @@ impl Interpreter for AlterTableClusterKeyInterpreter {
             .get_table(tenant.as_str(), &plan.database, &plan.table)
             .await?;
 
```

Comment:
Why we check_modify for cluster operation?

Reply:
I am not quite sure, cause it is a type operation of `AlterTableStmt`, so i add `check_can_modify_table` in every `AlterTableStmt` operation, it forbid modify the table if it is a shared table.

@dantengsky have a look?

Reply:
I think we should add a new share access check like management mode check here: https://github.com/datafuselabs/databend/blob/f2a8f0da2e966376b954bf7e357ff0c4ce86704f/src/query/service/src/interpreters/access/management_mode_access.rs

Reply:
It seems we should refactoring the access in InterpreterFactory to a common access instead the ManagementAccess.
Access.check():
management_access.check()?(
shared_table_access.check()?

Reply:
Open a PR for it in https://github.com/datafuselabs/databend/pull/7687
Then we can register a new checker in Accessor for the shared_table_accessor.